tests/unit/test-class-wc-payment-gateway-wcpay-subscriptions-payment-method-order-note.php

```diff
@@ -75,6 +75,16 @@ class WC_Payment_Gateway_WCPay_Subscriptions_Payment_Method_Order_Note_Test exte
 	 */
 	private $mock_wcpay_account;
 
+	public function tear_down() {
+		parent::tear_down();
+
+		// So that the gateway hooks are re-attached for the next test, we need to reset the flag that prevents the integration hooks from being attached.
+		$reflection = new \ReflectionClass( $this->wcpay_gateway );
+		$property   = $reflection->getProperty( 'has_attached_integration_hooks' );
+		$property->setAccessible( true );
+		$property->setValue( $this->wcpay_gateway, false );
+	}
```

Comment:
Any ideas on a better way to handle this? 

Because the `$has_attached_integration_hooks` private static variable is set to `true` when the `maybe_init_subscriptions()` is called and the callbacks are attached, we need to make sure we unset that flag so that the next `WC_Payment_Gateway_WCPay` instance will attach the callbacks again.

Reply:
I'm no expert at unit tests, so not really. Maybe just recreating the test gateway object would be easier to read?

Something like `$this->mock_gateway = new WC_Payments_Gateway( /* ... */ );` might be enough to do the trick 🤔

Reply:
We need to use a static variable in this case because the `$has_attached_integration_hooks` state needs to be shared across all instances of the trait. 

This is essentially the set up in a rough example. I've created a little example here: https://onlinephp.io/c/f87e6 

Essentially, by using a static variable, all payment method instances will inherit whether any of them have attached the callbacks. By using a standard variable, they wouldn't inherit that knowledge from the other payment method objects.